### The objective of this competition is to predict the probability that a customer does not pay back their credit card balance amount in the future based on their monthly customer profile. The target binary variable is calculated by observing 18 months performance window after the latest credit card statement, and if the customer does not pay due amount in 120 days after their latest statement date it is considered a default event.

The dataset contains aggregated profile features for each customer at each statement date. Features are anonymized and normalized, and fall into the following general categories:

    D_* = Delinquency variables
    S_* = Spend variables
    P_* = Payment variables
    B_* = Balance variables
    R_* = Risk variables

## For categorical feature analysis, it just used as exercise, because the result with the combined feature below is worse. So I don't use it in training model.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from matplotlib.pyplot import figure

In [2]:
df_train = pd.read_parquet("../input/amex-parquet/train_data.parquet")

According to powerbi interaction analysis, I choose the following combined categorical features.

In [3]:
categorical_features = ['customer_ID','B_30', 'B_38', 'D_63', 'D_64', 'D_66', 'D_68', 'D_114', 'D_116', 'D_117', 'D_120', 'D_126', 'target']
df_train[categorical_features] = df_train[categorical_features].astype("str")
df_categ = df_train[categorical_features]
    
B30 = df_categ[df_categ.B_30=='0.0']
B38 = df_categ[(df_categ.B_38=='1.0') | (df_categ.B_38=='2.0')]
D = df_categ[(df_categ.D_114=='1.0') & (df_categ.D_64=='O') & (df_categ.D_68=='6.0')]
U = B30.loc[B30.index & B38.index & D.index]  
U['pre_target']=0
    

print(U[['customer_ID','target','pre_target']])

/opt/conda/lib/python3.7/site-packages/ipykernel_launcher.py:8: FutureWarning: Index.__and__ operating as a set operation is deprecated, in the future this will be a logical operation matching Series.__and__.  Use index.intersection(other) instead
  


                                               customer_ID target  pre_target
0        0000099d6bd597052cdcda90ffabf56573fe9d7c79be5f...      0           0
1        0000099d6bd597052cdcda90ffabf56573fe9d7c79be5f...      0           0
2        0000099d6bd597052cdcda90ffabf56573fe9d7c79be5f...      0           0
3        0000099d6bd597052cdcda90ffabf56573fe9d7c79be5f...      0           0
4        0000099d6bd597052cdcda90ffabf56573fe9d7c79be5f...      0           0
...                                                    ...    ...         ...
5531354  fffee847c5c1af7dbdd36d98fea882893256c422cde86c...      0           0
5531356  fffee847c5c1af7dbdd36d98fea882893256c422cde86c...      0           0
5531357  fffee847c5c1af7dbdd36d98fea882893256c422cde86c...      0           0
5531358  fffee847c5c1af7dbdd36d98fea882893256c422cde86c...      0           0
5531359  fffee847c5c1af7dbdd36d98fea882893256c422cde86c...      0           0

[1035706 rows x 3 columns]


In [4]:
U.target.value_counts()/U.shape[0]

0    0.970569
1    0.029431
Name: target, dtype: float64

For these categorical features: B_30('0.0'), B_38('1.0' or '2.0'), D_114('1.0'), D_64('O'), and D_68('6.0'), if samples have the all former information, we set their predicted target as 0. The accuracy of 1035706 samples is 0.97. 